test/e2e/common.go

```diff
@@ -61,13 +62,31 @@ func Byf(format string, a ...interface{}) {
 	By(fmt.Sprintf(format, a...))
 }
 
-func setupSpecNamespace(ctx context.Context, specName string, clusterProxy framework.ClusterProxy, artifactFolder string) (*corev1.Namespace, context.CancelFunc) {
-	Byf("Creating a namespace for hosting the %q test spec", specName)
+func setupSpecNamespace(ctx context.Context, namespaceName string, clusterProxy framework.ClusterProxy, artifactFolder string) (*corev1.Namespace, context.CancelFunc) {
+	Byf("Creating a namespace for hosting the %q test spec", namespaceName)
+	logpath := filepath.Join(artifactFolder, "clusters", clusterProxy.GetName())
+	namespace, err := e2e_namespace.Get(ctx, clusterProxy.GetClientSet(), namespaceName)
+	if err == nil {
+		// namespace exists wire it up
+		Byf("Creating event watcher for existing namespace %q", namespace.Name)
+		watchesCtx, cancelWatches := context.WithCancel(ctx)
+		go func() {
+			defer GinkgoRecover()
+			framework.WatchNamespaceEvents(watchesCtx, framework.WatchNamespaceEventsInput{
+				ClientSet: clusterProxy.GetClientSet(),
+				Name:      namespace.Name,
+				LogFolder: logpath,
+			})
+		}()
+
+		return namespace, cancelWatches
+	}
```

Comment:
In case of an error, we should exit unless it's an API server not found error

Reply:
good catch